specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json

```diff
@@ -1932,113 +2063,133 @@
       "description": "Describes metadata for retrieving price info.",
-      "required": [
-        "meterId",
-        "meterType"
-      ],
       "type": "object",
       "properties": {
         "meterId": {
           "description": "Meter id of the Sku.",
-          "type": "string"
+          "type": "string",
+          "readOnly": true
         },
         "meterType": {
           "description": "The type of the meter.",
-          "type": "string"
+          "type": "string",
+          "readOnly": true
         }
       }
     },
     "SkuInformation": {
       "description": "Information of the sku.",
-      "required": [
-        "sku",
-        "enabled",
-        "properties"
-      ],
       "type": "object",
       "properties": {
         "sku": {
           "$ref": "#/definitions/Sku",
-          "description": "The Sku."
+          "description": "The Sku.",
+          "readOnly": true
         },
         "enabled": {
           "description": "The sku is enabled or not.",
-          "type": "boolean"
+          "type": "boolean",
+          "readOnly": true
         },
         "properties": {
           "$ref": "#/definitions/SkuProperties",
           "description": "Properties of the sku.",
+          "readOnly": true,
           "x-ms-client-flatten": true
         }
       }
     },
     "SkuProperties": {
       "description": "Properties of the sku.",
-      "required": [
-        "capacity",
-        "costs",
-        "apiVersions"
-      ],
       "type": "object",
       "properties": {
         "destinationToServiceLocationMap": {
           "description": "The map of destination location to service location.",
           "type": "array",
           "items": {
             "$ref": "#/definitions/DestinationToServiceLocationMap"
-          }
+          },
+          "readOnly": true
         },
         "capacity": {
           "$ref": "#/definitions/SkuCapacity",
-          "description": "Capacity of the Sku."
+          "description": "Capacity of the Sku.",
+          "readOnly": true
         },
         "costs": {
           "description": "Cost of the Sku.",
           "type": "array",
           "items": {
             "$ref": "#/definitions/SkuCost"
-          }
+          },
+          "readOnly": true
         },
         "apiVersions": {
           "description": "Api versions that support this Sku.",
           "type": "array",
           "items": {
             "type": "string"
-          }
+          },
+          "readOnly": true
         },
         "disabledReason": {
           "description": "Reason why the Sku is disabled.",
-          "type": "string"
+          "enum": [
+            "None",
+            "Country",
+            "Region",
+            "Feature",
+            "OfferType"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "SkuDisabledReason",
+            "modelAsString": false
+          }
+        },
```

Comment:
You can always provide description of enum values as below:

```json
        "disabledReason": {
          "type": "string",
          "description": "Reason why the Sku is disabled.",
          "enum": [
            "None",
            "Country",
            "Region",
            "Feature",
            "OfferType",
            "Bitrate"
          ],
          "x-ms-enum": {
            "name": "SkuDisabledReason",
            "values": [
              {
                "value": "None",
                "description": "disabled due to unknown reason"
              },
              {
                "value": "Country",
                "description": "SKU is not available in the country."
              },
              {
                "value": "Feature",
                "description": "Feature is not enabled for the SKU."
              },
              ...
            ],
            "modelAsExtensible": false
          },
        },
```

Reply:
nice to learn comment taken